eth/db/account.py

```diff
@@ -435,6 +459,27 @@ def persist(self) -> None:
             self._batchdb.commit_to(write_batch, apply_deletes=False)
         self._root_hash_at_last_persist = new_root_hash
 
+        return witness_index
+
+    def _get_accessed_node_hashes(self) -> Set[Hash32]:
+        return cast(Set[Hash32], self._raw_store_db.keys_read)
+
+    @to_dict
+    def _get_witness_metadata(self) -> Iterable[Tuple[Address, Tuple[bool, Tuple[int, ...]]]]:
+        for address in self._accessed_accounts:
+            did_access_bytecode = address in self._accessed_bytecodes
+            if address in self._account_stores:
+                accessed_storage_slots = self._account_stores[address].get_accessed_slots()
+            else:
+                accessed_storage_slots = tuple()
+            yield address, (did_access_bytecode, accessed_storage_slots)
```

Comment:
Thoughts on making this a `NamedTuple`?.  I think it would improve the typing, making the types have names and field names.